src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs

```diff
@@ -110,7 +110,10 @@ protected IDictionary CapNames
         internal RegexCode _code;                           // if interpreted, this is the code for RegexInterpreter
         internal bool _refsInitialized = false;
 
-        internal static LinkedList<CachedCodeEntry> s_livecode = new LinkedList<CachedCodeEntry>();// the cache of code and factories that are currently loaded
+        // the cache of code and factories that are currently loaded:
+        internal static Dictionary<CachedCodeEntryKey, CachedCodeEntry> s_livecode = new Dictionary<CachedCodeEntryKey, CachedCodeEntry>();
```

Comment:
Since the default is going to be very common, you probably want to pre size:
` internal static Dictionary<CachedCodeEntryKey, CachedCodeEntry> s_livecode = new Dictionary<CachedCodeEntryKey, CachedCodeEntry>(s_cacheSize);`